docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+The following part of the walkthrough explains how you can set up your agents in order to collect metrics from the applications running on your cluster.
+Those metrics will then be available for you to autoscale the resources of your cluster, via the DCA.
+If your agents are already instrumented and configured to communicate with the DCA, you can directly jump to the [running the hpa]() section.
+
+
+## Running the HPA
+
+At this point, you should be seeing:
+```
+PODS
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-agent-4c5pp                      1/1       Running   0          14m
+default       datadog-agent-ww2da                      1/1       Running   0          14m
+default       datadog-agent-2qqd3                      1/1       Running   0          14m
+[...]
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          16m
+default       nginx-6757dd8769-5xzp2                   1/1       Running   0          3m
+
+```
+
+Now is time to create a Horizontal Pod Autoscaler manifest. If you take a look at /manifest/cluster-agent/hpa-example/hpa-manifest.yaml, you will see:
+- The HPA is configured to autoscale the Deployment called nginx
+- The maximum number of replicas created will be 5 and the minumum is 1
+- The metric used is `nginx.net.request_per_s` and the scope is `kube_container_name: nginx`. Note that this metric format corresponds to the Datadog one.
+
+Every 30 seconds (this can be configured) Kubernetes will query the DCA to get the value of this metric and will autoscale proportionally if necessary.
+For advanced use cases, it is possible to have several metrics in the same HPA, as you can see [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-multiple-metrics) the largest of the proposed value will be the one chosen.
+
+Now, let's create our nginx deployment:
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/nginx.yaml`
+
+Then, let's apply the HPA
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/hpa-manifest.yaml`
+
+### Stressing your service
+
+Now is time to see the magic happening!
+
+If you curl the IP of the nginx service as follows:
+`curl <nginx_svc>:8090/nginx_status`, you should receive an output like:
+
+```
+$ curl 192.168.254.216:8090/nginx_status
+
+Active connections: 1 
+server accepts handled requests
+ 1 1 1 
+Reading: 0 Writing: 1 Waiting: 0 
+```
+
+Behind the scene, the number of request per second also increased. 
+This metric is being collected by the node agent, as it autodiscovered the nginx pod through its annotations (for more information on how autodiscovery works, see our doc [here](https://docs.datadoghq.com/agent/autodiscovery/#template-source-kubernetes-pod-annotations)).
+Therefore, if you stress it, you will see the uptick in your Datadog app. 
```

Comment:
r/`Therefore, if you stress it, you will see the uptick in your Datadog app.`/ `Therefore, if you stress it, you should see the uptick in your Datadog app.`